src/main/java/com/google/firebase/messaging/AndroidFcmOptions.java

```diff
@@ -0,0 +1,67 @@
+package com.google.firebase.messaging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.Key;
+import com.google.firebase.internal.FcmOptionsUtil;
+
+/**
+ * Represents the Android-specific FCM options that can be included in an {@link AndroidConfig}.
+ * Instances of this class are thread-safe and immutable.
+ */
+public class AndroidFcmOptions {
+
+  @Key("analytics_label")
+  private final String analyticsLabel;
+
+  private AndroidFcmOptions(Builder builder) {
+    checkArgument(FcmOptionsUtil.validAnalyticsLabelFormat(builder.analyticsLabel),
```

Comment:
This essentially makes the analyticsLabel required. Was that the intention? And what happens if analyticsLabel is null? You might want to check for that in your util.

Reply:
Wops, nice catch! Added test for this as well.